localstack/services/opensearch/provider.py

```diff
@@ -146,7 +147,7 @@ def create_cluster(
     # FIXME: in AWS, the Endpoint is set once the cluster is running, not before (like here), but our tests and
     #  in particular cloudformation currently relies on the assumption that it is set when the domain is created.
     status = region.opensearch_domains[domain_key.domain_name]
-    status["Endpoint"] = cluster.url.split("://")[-1]
+    status["Endpoint"] = cluster.url.split("://")[-1].replace("0.0.0.0", LOCALSTACK_HOSTNAME)
```

Comment:
I think the content of `cluster.url` here depends on the `EDGE_BIND_HOST` (as implemented in `cluster_manager.py`). So this won't work if the `EDGE_BIND_HOST` is not `0.0.0.0`, right?

Reply:
I assumed that for usability, if something else than 0.0.0.0 is given as the bind host, the user would then consciously choose the host and therefore want to have that as the output. In the case that it is bound to 0.0.0.0 it mostly means it runs in docker. 

We could introduce some better way to transform this (check host bindings of docker container if possible) but afaik there will be some changes around the variables surrounding hosts, so it might be better to save these optimizations for when that is fixed.

Reply:
Ah, okay. This reasoning is sound, but quite implicit. Could you add some comments there explaining the replace?